template-tags.php

```diff
@@ -676,19 +676,31 @@ function coauthors_wp_list_authors( $args = array() ) {
  * This is a replacement for using get_avatar(), which only operates on email addresses and cannot differentiate
  * between Guest Authors (who may share an email) and regular user accounts
  *
- * @param  object   $coauthor The Co Author or Guest Author object
- * @param  int      $size     The desired size
- * @return string             The image tag for the avatar, or an empty string if none could be determined
+ * @param  object        $coauthor The Co Author or Guest Author object.
+ * @param  int           $size     The desired size.
+ * @param  string        $default  Optional. URL for the default image or a default type. Accepts '404'
+ *                                 (return a 404 instead of a default image), 'retro' (8bit), 'monsterid'
+ *                                 (monster), 'wavatar' (cartoon face), 'indenticon' (the "quilt"),
+ *                                 'mystery', 'mm', or 'mysteryman' (The Oyster Man), 'blank' (transparent GIF),
+ *                                 or 'gravatar_default' (the Gravatar logo). Default is the value of the
+ *                                 'avatar_default' option, with a fallback of 'mystery'.
+ * @param  string        $alt      Optional. Alternative text to use in &lt;img&gt; tag. Default false.
+ * @param  array|string  $class    Array or string of additional classes to add to the &lt;img&gt; element. Default null.
+ * @return string                  The image tag for the avatar, or an empty string if none could be determined.
  */
-function coauthors_get_avatar( $coauthor, $size = 32, $default = '', $alt = false ) {
+function coauthors_get_avatar( $coauthor, $size = 32, $default = '', $alt = false, $class = null ) {
 	global $coauthors_plus;
 
 	if ( ! is_object( $coauthor ) ) {
 		return '';
 	}
 
+	if ( ! empty( $class ) && is_array( $class ) ) {
```

Comment:
What if `$class` is passed in as a string datatype?  Since I see in the PHPDocs for this function, string is allowed:

> @param  array|string  $class    Array or string of additional classes to add to the &lt;img&gt; element. Default null.

Reply:
If `$class` is passed as a string, we don't have to convert it to anything. Just pass it as-is to `get_guest_author_thumbnail()` or `get_avatar()`. That string will then be appended to the existing classes in those functions.

Reply:
Do you think it would be better to move this check into `get_guest_author_thumbnail()`, so then we can pass in `$class` as either an `array` or `string` datatype?

Reply:
Yes, I agree. That way we can pass in the $class parameter as is, and `get_guest_author_thumbnail()` and `get_avatar()` will check if it's an array or string.